vault/plugin_catalog.go

```diff
@@ -140,6 +143,37 @@ func (p *pluginClient) Conn() grpc.ClientConnInterface {
 	return p.clientConn
 }
 
+func (p *pluginClient) Reload() error {
+	p.logger.Debug("reload external plugin process")
+	return p.reloadFunc()
+}
+
+// reloadExternalPlugin
+// This should be called with the write lock held.
```

Comment:
Not a big deal either way, but why not take the lock within this function?

Reply:
I was following a similar pattern that I used for cleanupExternalPlugin. If this method is ever called by the plugin catalog with the lock held then we will deadlock. I am not sure if we would ever need that though.